src/main/java/com/minecolonies/coremod/entity/pathfinding/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -244,6 +279,19 @@ else if (pathResult.getStatus() == PathFindingStatus.CALCULATION_COMPLETE)
         stuckHandler.checkStuck(this);
     }
 
+    /**
+     * Similar to WalkNodeProcessor.getGroundY but not broken.
+     * @param world the world.
+     * @param pos the position to check.
+     * @return the next y level to go to.
+     */
+    public static double getSmartGroundY(final IBlockReader world, final BlockPos pos)
+    {
+        final BlockPos blockpos = pos.down();
+        final VoxelShape voxelshape = world.getBlockState(blockpos).getCollisionShape(world, blockpos);
+        return (double)blockpos.getY() + (voxelshape.isEmpty() ? 1.0D : voxelshape.getEnd(Direction.Axis.Y));
+    }
+
```

Comment:
Again document why the original is broken.
See above.

Reply:
done